backbone.js

```diff
@@ -1447,16 +1457,12 @@
         name = '';
       }
       if (!callback) callback = this[name];
-      var router = this;
-      Backbone.history.route(route, function(fragment) {
-        var args = router._extractParameters(route, fragment);
-        if (router.execute(callback, args, name) !== false) {
-          router.trigger.apply(router, ['route:' + name].concat(args));
-          router.trigger('route', name, args);
-          Backbone.history.trigger('route', router, name, args);
-        }
-      });
-      return this;
+      var handler = {
+        route: route,
+        callback: callback,
+        name: name
+      };
+      Router.handlers.push(handler);
```

Comment:
*If two routers register the same route, why not have both fire there callbacks?*

Shrug. Dunno. I would expect it to behave that way, since the router doesn't really *do* all that much. I think the argument might be something along the lines of..."each route should only have 1 thing happen. so you only need one callback, ever, even if you have multiple routers." But I don't really think this limitation is going to be particularly useful to anyone making a webapp. I just kept the behavior to keep it as BC as possible.